Structural/Flyweight/CharacterFlyweight.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace DesignPatterns\Structural\Flyweight;
+
+/**
+ * Implements the flyweight interface and adds storage for intrinsic state, if any.
+ * Instances of concrete flyweights are shared by means of a factory.
+ */
+class CharacterFlyweight implements FlyweightInterface
+{
+    /**
+     * Any state stored by the concrete flyweight must be independent of its context.
+     * For flyweights representing characters, this is usually the corresponding character code.
+     * @var string
+     */
+    private $name;
+
+    /**
+     * Constructor.
+     * @param string $name
+     */
+    public function __construct($name) {
```

Comment:
did you run [PHP CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer)? I think there are some coding style issues here.